app/main.py

```diff
@@ -1,34 +1,57 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __eq__(self, other: tuple) -> bool:
+        row, column = other
+        if self.row == row and self.column == column:
+            return True
+        return False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+        self,
+        start: tuple[int],
+        end: tuple[int],
+        is_drowned: bool = False
+    ) -> None:
+        if start[0] == end[0]:
+            self.decks = [
+                Deck(start[0], i) for i in range(start[1], end[1] + 1)
+            ]
+        else:
+            self.decks = [
+                Deck(i, start[1]) for i in range(start[0], end[0] + 1)
+            ]
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck | None:
+        for deck in self.decks:
+            if (row, column) == deck:
+                return deck
+        return None
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+        if all([not deck.is_alive for deck in self.decks]):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: tuple[tuple[int]]) -> None:
+        self.ships = [Ship(ship[0], ship[1]) for ship in ships]
+        self.battlefield = [["~" for _ in range(10)] for _ in range(10)]
```

Comment:
why you need it?

if you want to implement the battlefield print method, put this logic in a separate method